<a href="https://colab.research.google.com/github/maififteen/google-colab-backup/blob/main/Salinan_Penelitian_Deteksi_Masker_2021_UNDIP.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Initialize Git Repository  
Untuk dapat melihat riwayat perubahan yang terjadi selama proses penelitian, project dimasukkan ke dalam github.  

In [22]:
### Check Operating System
!apt install neofetch

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following package was automatically installed and is no longer required:
  libnvidia-common-460
Use 'apt autoremove' to remove it.
The following NEW packages will be installed:
  neofetch
0 upgraded, 1 newly installed, 0 to remove and 40 not upgraded.
Need to get 74.8 kB of archives.
After this operation, 365 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 neofetch all 3.4.0-1 [74.8 kB]
Fetched 74.8 kB in 0s (1,035 kB/s)
Selecting previously unselected package neofetch.
(Reading database ... 148486 files and directories currently installed.)
Preparing to unpack .../neofetch_3.4.0-1_all.deb ...
Unpacking neofetch (3.4.0-1) ...
Setting up neofetch (3.4.0-1) ...
Processing triggers for man-db (2.8.3-2ubuntu0.1) ...


In [23]:
!neofetch

            .-/+oossssoo+/-. 
        `:+ssssssssssssssssss+:` 
      -+ssssssssssssssssssyyssss+- 
    .ossssssssssssssssssdMMMNysssso. 
   /ssssssssssshdmmNNmmyNMMMMhssssss/ 
  +ssssssssshmydMMMMMMMNddddyssssssss+ 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/ 
.ssssssssdMMMNhsssssssssshNMMMdssssssss. 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+ 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+ 
.ssssssssdMMMNhsssssssssshNMMMdssssssss. 
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/ 
  +sssssssssdmydMMMMMMMMddddyssssssss+ 
   /ssssssssssshdmNNNNmyNMMMMhssssss/ 
    .ossssssssssssssssssdMMMNysssso. 
      -+sssssssssssssssssyyyssss+- 
        `:+ssssssssssssssssss+:` 
            .-/+oossssoo+/-. 
root@df1a455f5305 
----------------- 
OS: Ubuntu 18.04.5 LTS x86_64 
Host: Google Compute Engine 
Kernel: 5.4.104+ 
Uptime: 1 hour, 13 mins 
Packages: 1347 
Shell: bash 4.4.20 
Terminal: jupyter-noteboo 
CPU: Intel Xeon (2) @ 2.299G

### Mount Google Drive  
Sebelum melakukan mounting Google Drive dengan host Google Colab, dataset [MaskedFace-Net](https://github.com/cabani/MaskedFace-Net) ditambahkan atau disimpan ke dalam direktori Google Drive akun pribadi. Mounting dengan perintah berikut.

In [24]:
from google.colab import drive
drive.mount('/content/drive', force_remount=True)

Mounted at /content/drive


### Unzip The Dataset  
Terdapat 3 kategori dataset yaitu CMFD (Correctly Masked Face Dataset), IMFD (Incorrectly Masked Face Dataset), dan FFHQ. Hasil ektraksi masing-masing folder akan masuk ke dalam folder `../dataframe`.  

#### Extract the CMFD
CMFD terdiri dari dua file zip. Ektraksi dilakukan terhadap satu filenya terlebih dahulu yaitu `CMFD.zip`. 

Note: Kode di bawah mengekstrak file CMFD.zip dari Shareddrives. Perhatikan jika akan menjalankan sel dibawah ubah `path source` ke `/content/drive/MyDrive/Colab/MaskDetection/Dataset/CMFD/CMFD.zip`.

In [ ]:
!unzip "/content/drive/MyDrive/Colab/MaskDetection/Dataset/CMFD/CMFD.zip" -d "/content/drive/MyDrive/Colab/MaskDetection/Dataframe/CMFD"

Output streaming akan dipotong hingga 5000 baris terakhir.
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/CMFD/29000/29796_Mask.jpg  
 extracting: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/CMFD/29000/29797_Mask.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/CMFD/29000/29798_Mask.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/CMFD/29000/29799_Mask.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/CMFD/29000/29800_Mask.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/CMFD/29000/29801_Mask.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/CMFD/29000/29802_Mask.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/CMFD/29000/29803_Mask.jpg  
 extracting: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/CMFD/29000/29804_Mask.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/CMFD/29000/29805_Mask.jpg  
 

#### Count Total Images in CMFD Directory  
Hitung jumlah gambar yang ada di folder hasil ektraksi file `CMFD.zip`.

In [25]:
base_dataframe = "/content/drive/MyDrive/Colab/MaskDetection/Dataframe"

In [26]:
import pathlib

cmfd_dir = base_dataframe + "/CMFD"
cmfd_dir = pathlib.Path(cmfd_dir)

In [27]:
cmfd_count = len(list(cmfd_dir.glob('*/*.jpg')))
print(cmfd_count)

15508


Terdapat 15508 gambar yang ada di folder `CMFD` hasil ekstraksi `CMFD.zip`.

#### Extract the IMFD  
IMFD terdiri dari dua file zip. Ektraksi dilakukan terhadap seluruh file yang ada di folder ini. 

In [ ]:
!unzip "/content/drive/MyDrive/Colab/MaskDetection/Dataset/IMFD/IMFD.zip" -d "/content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/"

Output streaming akan dipotong hingga 5000 baris terakhir.
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/29000/29811_Mask_Mouth_Chin.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/29000/29812_Mask_Mouth_Chin.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/29000/29813_Mask_Mouth_Chin.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/29000/29814_Mask_Mouth_Chin.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/29000/29815_Mask_Chin.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/29000/29816_Mask_Mouth_Chin.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/29000/29817_Mask_Mouth_Chin.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/29000/29818_Mask_Mouth_Chin.jpg  
 extracting: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/29000/29819_Mask_Mouth_Chin.jpg  
  

In [ ]:
!unzip "/content/drive/MyDrive/Colab/MaskDetection/Dataset/IMFD/IMFD1.zip" -d "/content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD"

Output streaming akan dipotong hingga 5000 baris terakhir.
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/64000/64738_Mask_Mouth_Chin.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/64000/64739_Mask_Chin.jpg  
 extracting: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/64000/64740_Mask_Mouth_Chin.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/64000/64741_Mask_Nose_Mouth.jpg  
 extracting: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/64000/64742_Mask_Mouth_Chin.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/64000/64743_Mask_Mouth_Chin.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/64000/64744_Mask_Mouth_Chin.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/64000/64745_Mask_Mouth_Chin.jpg  
  inflating: /content/drive/MyDrive/Colab/MaskDetection/Dataframe/IMFD/64000/64746_Mask_Mouth_Chin.jpg  
  

#### Count Total Images in IMFD Directory  
Hitung jumlah gambar yang ada di folder hasil ektraksi file `IMFD.zip`.

In [28]:
imfd_dir = base_dataframe + "/IMFD"
imfd_dir = pathlib.Path(imfd_dir)

imfd_count = len(list(imfd_dir.glob('*/*.jpg')))
print(imfd_count)

66734


Total gambar yang ada di folder hasil ektraksi file `IMFD.zip` adalah 66734.

#### Count Each Class in IMFD  
IMFD terdiri dari 3 kelas berbeda, yaitu:  
1. Uncovered Chin (Mask_Nose_Mouth)  
2. Uncovered Nose (Mask_Mouth_Chin)  
3. Uncovered Mouth and Nose (Mask_Chin)  

Hitung jumlah masing-masing data pada setiap kelas.

In [29]:
matching_patterns = [
                    '*/*Mask_Nose_Mouth.jpg',
                    '*/*Mask_Mouth_Chin.jpg',
                    '*/*Mask_Chin.jpg'
                  ]

In [30]:
sum = 0
for pattern in matching_patterns:
  count = len(list(imfd_dir.glob(pattern)))
  sum += count
  print(count)

print("Total images in IMFD: ", sum)

6245
55653
4836
Total images in IMFD:  66734


#### Extract the FFHQ  
FFHQ di pindahkan ke dalam folder `../Dataframe` agar siap digunakan.

#### Count Total Images in FFHQ Directory  
Hitung jumlah gambar yang ada di folder `../Dataframe/FFHQ`.

In [32]:
ffhq_dir = base_dataframe + "/FFHQ"
ffhq_dir = pathlib.Path(ffhq_dir)

ffhq_count = len(list(ffhq_dir.glob('*/*.jpg')))
print(ffhq_count)

0


### Next To Do
1. Ambil masing-masing 4836 dataset dari masing-masing kelas.
2. Buat folder untuk masing-masing kelas.
3. 